crates/starknet-server/tests/common/utils.rs

```diff
@@ -139,3 +140,47 @@ pub async fn send_ctrl_c_signal(process: &Child) {
         kill.wait().unwrap();
     }
 }
+
+/// Wrapper of file name which attempts to delete the file when the variable is dropped.
+/// Appends a random sequence to the file name base to make it unique.
+/// Prevents name collisions - no need to come up with unique names for files (e.g. when dumping).
+/// Automatically deletes the underlying file when the variable is dropped - no need to remember
+/// deleting.
+pub struct UniqueAutoDeletableFile {
+    pub path: String,
+}
+
+impl UniqueAutoDeletableFile {
+    /// Appends a random sequence to the name_base to make it unique
+    /// Unlike [NamedTempFile](https://docs.rs/tempfile/latest/tempfile/struct.NamedTempFile.html),
+    /// it doesn't create the file.
+    pub fn new(name_base: &str) -> Self {
+        Self { path: format!("{name_base}-{}", generate_u32_random_number()) }
+    }
+}
+
+impl Drop for UniqueAutoDeletableFile {
+    fn drop(&mut self) {
+        remove_file(&self.path)
+    }
+}
+
+#[cfg(test)]
+mod test_unique_auto_deletable_file {
+    use std::path::Path;
+
+    use super::UniqueAutoDeletableFile;
+
+    #[test]
+    fn foo() {
+        let file = UniqueAutoDeletableFile::new("foo");
```

Comment:
I think that one test should be added with 2 instances of `UniqueAutoDeletableFile` like:
```
let first_foo = UniqueAutoDeletableFile::new("foo");
let second_foo = UniqueAutoDeletableFile::new("foo");
```

to prove that collisions will be prevented

Reply:
Thanks for the suggestion. I added new tests in https://github.com/0xSpaceShard/starknet-devnet-rs/pull/253/commits/ba8de5bcad4f1fd4906e800203cce401b40b774e

Reply:
I would remove that loop, it's basically a test of `rand::thread_rng` it doesn't make sense in my opinion to test that in our code base